tensorboard/plugins/graph/graph_util.py

```diff
@@ -0,0 +1,88 @@
+# Copyright 2019 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""Utilities for graph plugin."""
+from tensorboard.compat.proto import function_pb2
+
+
+def _safe_copy_proto_list_values(to_proto_list, from_proto_list, get_key, error_msg):
+  """Safely copies the value from from_proto_list to to_proto_list.
+
+  Copies if to_proto_list does not contain an item with the same key. In case an item
+  is found with the same key, it checks whether contents match and, if not, it raises
+  a ValueError.
+
+  Args:
```

Comment:
These args don’t look right? Maybe:

```python
  """...

  Args:
    dst_proto_list: A `RepeatedCompositeContainer` or
      `RepeatedScalarContainer` into which values should be copied.
    src_proto_list: A container holding the same kind of values as in
      `dst_proto_list` from which values should be copied.
    get_key: A function that takes an element of `dst_proto_list` or
      `src_proto_list` and returns a key, such that if two elements have
      the same key then it is required that they be deep-equal. For
      instance, if `dst_proto_list` is a list of nodes, then `get_key`
      might be `lambda node: node.name` to indicate that if two nodes
      have the same name then they must be the same node. All keys must
      be hashable.
    error_msg: A user-friendly error message to be raised in the case
      that two nodes share a key but are distinct.
  """
```

Reply:
(If you don’t like the use of `Repeated*Container`, feel free to
substitute a more generic term like “proto repeated field container”.)